solidity/random-beacon/contracts/libraries/Groups.sol

```diff
@@ -342,41 +337,4 @@ library Groups {
 
         return shiftedIndex;
     }
```

Comment:
I would leave `setGroupMembers` untouched in this PR.

Reply:
@nkuba I'm not sure if we can leave it untouched based on the proposed solution in #2756 
> Instead of storing group member identifiers as uint32[] on-chain, we could store a hash of group member identifiers array and expect uint32[] identifier array to be submitted for every function requiring it.

and later
> `DkgResultApproved` should contain uint32[] members field that should have the final value (after eliminating misbehaving IA/DQ members)

My understanding is that we need to filter out IA/DQ before we set a hash of members that participated in dkg. `DkgResultApproved` will also emit members without IA/DQ members. In the following functions: `submitRelayEntry`, `reportRelayEntryTimeout`, `reportUnauthorizedSigning`, `notifyFailedHeartbeat` one of the params will be an array of members that participated in dkg that was emitted in `DkgResultApproved`.

Here's what this function can look like after the change: 1c429b3 I didn't test it yet, but the idea is to hash only those members that participated in dkg (without IA/DQ)